src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1262,19 +1269,20 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
                 throw new NotSupportedException(WorkspacesResources.Changing_document_property_is_not_supported);
             }
 
-            var changedDocumentIds = projectChanges.GetChangedDocuments(onlyGetDocumentsWithTextChanges: true).ToImmutableArray();
+            var changedDocumentIds = projectChanges.GetChangedDocuments(onlyGetDocumentsWithTextChanges: true, IgnoreUnchangeableDocumentsWhenApplyingChanges).ToImmutableArray();
 
             if (!this.CanApplyChange(ApplyChangesKind.ChangeDocument) && changedDocumentIds.Length > 0)
             {
                 throw new NotSupportedException(WorkspacesResources.Changing_documents_is_not_supported);
             }
 
-            foreach (var changedDocumentId in changedDocumentIds)
+            // Checking for unchangeable documents will only be done if we were asked not to ignore them.
+            foreach (var documentId in changedDocumentIds)
             {
-                var changedDocument = projectChanges.OldProject.GetDocumentState(changedDocumentId) ?? projectChanges.NewProject.GetDocumentState(changedDocumentId)!;
-                if (!changedDocument.CanApplyChange())
+                var document = projectChanges.OldProject.GetDocumentState(documentId) ?? projectChanges.NewProject.GetDocumentState(documentId)!;
```

Comment:
Unrelated to your change but any idea why we're looking at both OldProject and NewProject here? It's a changed document, so won't the first always return non-null and the stuff after ?? never run?

Reply:
I don't know, and I can't fins any comments on this in the [original PR](https://github.com/dotnet/roslyn/pull/30733). Any idea @heejaechang?

Reply:
can't remember why I added ?? :) but what @jasonmalinowski said is right, GetChangedDocument returns documentIds that only exist in both old and new snapshot so it should always exist in old snapshot.

Reply:
Thanks Heejae, I will clean this up .